__tests__/app.test.js

```diff
@@ -155,8 +154,183 @@ describe("app", () => {
+         
+            expect(result.body[0]).toHaveProperty("article_id");
+            expect(result.body[0]).toHaveProperty("title");
+            expect(result.body[0]).toHaveProperty("topic");
+            expect(result.body[0]).toHaveProperty("author");
+            expect(result.body[0]).toHaveProperty("body");
+            expect(result.body[0]).toHaveProperty("created_at");
+            expect(result.body[0]).toHaveProperty("votes");
+            expect(result.body[0]).toHaveProperty("article_img_url");
+        });
+    });
+  
+    test("It Returns 200 status and an array of objects for a particular article with the updated votes property", () => {
+      const article_id = 1;
+      const incremnt = { inc_votes: 50 };
+      return request(app)
+        .patch(`/api/articles/${article_id}`)
+        .send(incremnt)
+        .expect(200)
+        .then((result) => {
+          expect(result.body[0].votes).toBe(150);
+        
+        });
+    });
+  
+    test("It Returns 200 status and an array of objects for a particular article with the updated votes property", () => {
+      const article_id = 9;
+      const incremnt = { inc_votes: -50 };
+      return request(app)
+        .patch(`/api/articles/${article_id}`)
+        .send(incremnt)
+        .expect(200)
+        .then((result) => {
+          expect(result.body[0].votes).toBe(-50);
+        
+        });
+    });
+  
+    test('Returns Error with 404 when the article id it does not exist', () => {
+      const article_id = 50;
+      return request(app)
+      .patch(`/api/articles/${article_id}`)
+      .expect(404)
+      .then(({body})=>{
+      expect(body.msg).toBe("Article not found")
+  
+        });
+      
+    });
+  
+    test("it returns a 400 status with a message : (Invalid id) if the endpoint is invalid", () => {
+      const article_id = "Waled";
+      return request(app)
+        .patch(`/api/articles/${article_id}`)
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Invalid id");
+        });
+    });
   });
 
+  describe("9. GET /api/users", () => {
+
+    test("it returns a 200 status and  an array of 4 objects and every object has 3 proberties", () => {
+      return request(app)
+        .get("/api/users")
+        .expect(200)
+        .then((users) => {
+        expect(users.body).toHaveLength(4)
+        users.body.forEach((user) => {
+        expect(Object.keys(user).length).toBe(3);;
+        });
+        });
+    });
   
+    test("Returns an array of objects, where each object should have username, name and avatar_url proberties  ", () => {
+      return request(app)
+        .get("/api/users")
+        .expect(200)
+        .then((users) => {
+        
+            users.body.forEach((user) => {
+              expect(user).toHaveProperty("username");
+              expect(user).toHaveProperty("name");
+              expect(user).toHaveProperty("avatar_url");
+            });
+          
+        });
+    });
```

Comment:
These two test are very similar the only difference being the length is asserted in one but not the other, i would recommend combining these into one instead of having this as two seperate tests.